site/app/models/gradeable/TaGradedGradeable.php

```diff
@@ -335,6 +357,35 @@ public function clearDeletedGradedComponents() {
         $this->deleted_graded_components = [];
     }
 
+    /**
+     * Sets the overall comment for a grader. Access should be checked before calling this function.
+     * @param string $comment. The comment to be saved.
+     * @param string $grader_id. The grader that made the comment.
+     */
+    public function setOverallComment($comment, $grader_id) {
+        $this->overall_comment[$grader_id] = $comment;
+    }
+
+    /**
+     * Retrieves a mapping of grader id to overall comment. If grader is passed in, returns only
+     * the key, value pair for that grader.
+     * @param User|null $grader The grader to retrieve a comment for. Optional.
+     */
+    public function getOverallComment(User $grader = null) {
+        if ($grader === null) {
+            return $this->overall_comment;
+        }
+        else {
+            if (array_key_exists($grader->getId(), $this->overall_comment)) {
+                return array($grader->getId() => $this->overall_comment[$grader->getId()]);
```

Comment:
Can we change this to use the `[]` array instantiation here

`[ $grader->getId() => $this->overall_comment[$grader->getId()] ]`

Reply:
Updated